Sources/Foundation/Decimal.swift

```diff
@@ -2242,19 +2242,19 @@ extension Decimal {
         _exponent = newExponent
         return .noError
     }
-    fileprivate mutating func power(_ p:UInt, roundingMode:RoundingMode) -> CalculationError {
+    fileprivate mutating func power(_ p:Int, roundingMode:RoundingMode) -> CalculationError {
         if isNaN {
             return .overflow
         }
-        var power = p
+        var power = abs(p)
```

Comment:
This does become a bit confusing.  It would've been nicer to use a more verbose name (e.g. `absPower`).

Reply:
You're right, I've changed the name to `absPower`.